internal/http/server.go

```diff
@@ -75,18 +75,32 @@ func NewServer(log logger.Logger, config *config.AppConfig, sse *sse.Server, db
 
 func (s Server) Open() error {
 	addr := fmt.Sprintf("%v:%v", s.config.Config.Host, s.config.Config.Port)
-	listener, err := net.Listen("tcp", addr)
+
+	var err error
+	for _, proto := range []string{"tcp", "tcp4", "tcp6"} {
+		if err = s.tryToServe(addr, proto); err == nil {
+			return err
+		}
```

Comment:
Shouldn't this be `; err != nil` so it can return the err, or have it return nil?

Reply:
most certainly.

Reply:
actually... no, that was intentional. we can force it to nil, just an old habit.